core/src/main/java/hudson/util/RingBufferLogHandler.java

```diff
@@ -35,25 +35,19 @@
  */
 public class RingBufferLogHandler extends Handler {
 
-    private static final int DEFAULT_RING_BUFFER_SIZE = Integer.getInteger(RingBufferLogHandler.class.getName() + ".defaultSize", 256);
-
     private int start = 0;
     private final LogRecord[] records;
     private volatile int size = 0;
 
-    /**
-     * This constructor is deprecated. It can't access system properties with {@link jenkins.util.SystemProperties}
-     * as it's not legal to use it on remoting agents.
-     */
-    @Deprecated
-    public RingBufferLogHandler() {
-        this(DEFAULT_RING_BUFFER_SIZE);
-    }
-
+    //suggested size 256
     public RingBufferLogHandler(int ringSize) {
         records = new LogRecord[ringSize];
     }
 
+    public synchronized int getRecordsLength() {
```

Comment:
You can remove this method, it's returning the current length and it was not required by the Jira ticket.